### ISSUE TITLE
Update schemas in API

### ISSUE THREAD

**Issue by roed314** [LMFDB/lmfdb#4661]
The API can now show schemas, removing the need for the inventory.  In preparation for switching over, we should check the quality of the descriptions of the column data.

If you notice an issue, here's how to update the description of a column while logged in as editor (needs a fairly recent version of the LMFDB code).

```
sage: db.av_fq_isog.column_description("abvar_count", "the number of points on this abelian variety over the base field")
```

Alternatively, you can provide a dictionary:

```
sage: db.av_fq_isog.column_description(description={"abvar_count": "the number of points on this abelian variety over the base field", "q": "the cardinality of the base field"})
```

Once this is done, we can change the sidebar to link to the API rather than the inventory.

- [ ] [artin_field_data](https://beta.lmfdb.org/api/artin_field_data/)
- [ ] [artin_old2new_labels](https://beta.lmfdb.org/api/artin_old2new_labels/)
- [ ] [artin_reps](https://beta.lmfdb.org/api/artin_reps/)
- [ ] [av_fq_endalg_data](https://beta.lmfdb.org/api/av_fq_endalg_data/)
- [ ] [av_fq_endalg_factors](https://beta.lmfdb.org/api/av_fq_endalg_factors/)
- [ ] [av_fq_isog](https://beta.lmfdb.org/api/av_fq_isog/)
- [ ] [belyi_galmap_portraits](https://beta.lmfdb.org/api/belyi_galmap_portraits/)
- [ ] [belyi_galmaps](https://beta.lmfdb.org/api/belyi_galmaps/)
- [ ] [belyi_galmaps_prim](https://beta.lmfdb.org/api/belyi_galmaps_prim/)
- [ ] [belyi_passports](https://beta.lmfdb.org/api/belyi_passports/)
- [ ] [bmf_dims](https://beta.lmfdb.org/api/bmf_dims/)
- [ ] [bmf_forms](https://beta.lmfdb.org/api/bmf_forms/)
- [ ] [char_dir_orbits](https://beta.lmfdb.org/api/char_dir_orbits/)
- [ ] [char_dir_values](https://beta.lmfdb.org/api/char_dir_values/)
- [ ] [cluster_pictures](https://beta.lmfdb.org/api/cluster_pictures/)
- [ ] [ec_2adic](https://beta.lmfdb.org/api/ec_2adic/)
- [ ] [ec_classdata](https://beta.lmfdb.org/api/ec_classdata/)
- [ ] [ec_curvedata](https://beta.lmfdb.org/api/ec_curvedata/)
- [ ] [ec_curves](https://beta.lmfdb.org/api/ec_curves/)
- [ ] [ec_galrep](https://beta.lmfdb.org/api/ec_galrep/)
- [ ] [ec_iqf_labels](https://beta.lmfdb.org/api/ec_iqf_labels/)
- [ ] [ec_iwasawa](https://beta.lmfdb.org/api/ec_iwasawa/)
- [ ] [ec_localdata](https://beta.lmfdb.org/api/ec_localdata/)
- [ ] [ec_mwbsd](https://beta.lmfdb.org/api/ec_mwbsd/)
- [ ] [ec_nfcopy](https://beta.lmfdb.org/api/ec_nfcopy/)
- [ ] [ec_nfcurves](https://beta.lmfdb.org/api/ec_nfcurves/)
- [ ] [ec_padic](https://beta.lmfdb.org/api/ec_padic/)
- [ ] [ec_sympow](https://beta.lmfdb.org/api/ec_sympow/)
- [ ] [ec_torsion_growth](https://beta.lmfdb.org/api/ec_torsion_growth/)
- [ ] [fq_fields](https://beta.lmfdb.org/api/fq_fields/)
- [ ] [g2c_curves](https://beta.lmfdb.org/api/g2c_curves/)
- [ ] [g2c_endomorphisms](https://beta.lmfdb.org/api/g2c_endomorphisms/)
- [ ] [g2c_plots](https://beta.lmfdb.org/api/g2c_plots/)
- [ ] [g2c_ratpts](https://beta.lmfdb.org/api/g2c_ratpts/)
- [ ] [g2c_tamagawa](https://beta.lmfdb.org/api/g2c_tamagawa/)
- [ ] [gps_char](https://beta.lmfdb.org/api/gps_char/)
- [ ] [gps_crep](https://beta.lmfdb.org/api/gps_crep/)
- [ ] [gps_families](https://beta.lmfdb.org/api/gps_families/)
- [ ] [gps_gmodules](https://beta.lmfdb.org/api/gps_gmodules/)
- [ ] [gps_groups](https://beta.lmfdb.org/api/gps_groups/)
- [ ] [gps_groups_cc](https://beta.lmfdb.org/api/gps_groups_cc/)
- [ ] [gps_images](https://beta.lmfdb.org/api/gps_images/)
- [ ] [gps_qchar](https://beta.lmfdb.org/api/gps_qchar/)
- [ ] [gps_qrep](https://beta.lmfdb.org/api/gps_qrep/)
- [ ] [gps_small](https://beta.lmfdb.org/api/gps_small/)
- [ ] [gps_special_names](https://beta.lmfdb.org/api/gps_special_names/)
- [ ] [gps_st](https://beta.lmfdb.org/api/gps_st/)
- [ ] [gps_st0](https://beta.lmfdb.org/api/gps_st0/)
- [ ] [gps_subgroups](https://beta.lmfdb.org/api/gps_subgroups/)
- [ ] [gps_transitive](https://beta.lmfdb.org/api/gps_transitive/)
- [ ] [gps_zrep](https://beta.lmfdb.org/api/gps_zrep/)
- [ ] [halfmf_forms](https://beta.lmfdb.org/api/halfmf_forms/)
- [ ] [hecke_algebras](https://beta.lmfdb.org/api/hecke_algebras/)
- [ ] [hecke_ladic](https://beta.lmfdb.org/api/hecke_ladic/)
- [ ] [hecke_orbits](https://beta.lmfdb.org/api/hecke_orbits/)
- [ ] [hgcwa_complete](https://beta.lmfdb.org/api/hgcwa_complete/)
- [ ] [hgcwa_passports](https://beta.lmfdb.org/api/hgcwa_passports/)
- [ ] [hgcwa_unique_groups](https://beta.lmfdb.org/api/hgcwa_unique_groups/)
- [ ] [hgm_families](https://beta.lmfdb.org/api/hgm_families/)
- [ ] [hgm_motives](https://beta.lmfdb.org/api/hgm_motives/)
- [ ] [hmf_fields](https://beta.lmfdb.org/api/hmf_fields/)
- [ ] [hmf_forms](https://beta.lmfdb.org/api/hmf_forms/)
- [ ] [hmf_hecke](https://beta.lmfdb.org/api/hmf_hecke/)
- [ ] [lat_lattices](https://beta.lmfdb.org/api/lat_lattices/)
- [ ] [lf_fields](https://beta.lmfdb.org/api/lf_fields/)
- [ ] [lfunc_instances](https://beta.lmfdb.org/api/lfunc_instances/)
- [ ] [lfunc_lfunctions](https://beta.lmfdb.org/api/lfunc_lfunctions/)
- [ ] [lfunc_rs_knowls](https://beta.lmfdb.org/api/lfunc_rs_knowls/)
- [ ] [lfunc_search](https://beta.lmfdb.org/api/lfunc_search/)
- [ ] [lfunc_zeros](https://beta.lmfdb.org/api/lfunc_zeros/)
- [ ] [maass_newforms](https://beta.lmfdb.org/api/maass_newforms/)
- [ ] [maass_portraits](https://beta.lmfdb.org/api/maass_portraits/)
- [ ] [mf_boxes](https://beta.lmfdb.org/api/mf_boxes/)
- [ ] [mf_gamma1](https://beta.lmfdb.org/api/mf_gamma1/)
- [ ] [mf_gamma1_portraits](https://beta.lmfdb.org/api/mf_gamma1_portraits/)
- [ ] [mf_gamma1_subspaces](https://beta.lmfdb.org/api/mf_gamma1_subspaces/)
- [ ] [mf_hecke_cc](https://beta.lmfdb.org/api/mf_hecke_cc/)
- [ ] [mf_hecke_charpolys](https://beta.lmfdb.org/api/mf_hecke_charpolys/)
- [ ] [mf_hecke_lpolys](https://beta.lmfdb.org/api/mf_hecke_lpolys/)
- [ ] [mf_hecke_newspace_traces](https://beta.lmfdb.org/api/mf_hecke_newspace_traces/)
- [ ] [mf_hecke_nf](https://beta.lmfdb.org/api/mf_hecke_nf/)
- [ ] [mf_hecke_traces](https://beta.lmfdb.org/api/mf_hecke_traces/)
- [ ] [mf_newform_portraits](https://beta.lmfdb.org/api/mf_newform_portraits/)
- [ ] [mf_newforms](https://beta.lmfdb.org/api/mf_newforms/)
- [ ] [mf_newspace_portraits](https://beta.lmfdb.org/api/mf_newspace_portraits/)
- [ ] [mf_newspaces](https://beta.lmfdb.org/api/mf_newspaces/)
- [ ] [mf_subspaces](https://beta.lmfdb.org/api/mf_subspaces/)
- [ ] [mf_temp](https://beta.lmfdb.org/api/mf_temp/)
- [ ] [mf_twists_cc](https://beta.lmfdb.org/api/mf_twists_cc/)
- [ ] [mf_twists_nf](https://beta.lmfdb.org/api/mf_twists_nf/)
- [ ] [modlgal_reps](https://beta.lmfdb.org/api/modlgal_reps/)
- [ ] [modlmf_forms](https://beta.lmfdb.org/api/modlmf_forms/)
- [ ] [mwf_coeffs](https://beta.lmfdb.org/api/mwf_coeffs/)
- [ ] [mwf_forms](https://beta.lmfdb.org/api/mwf_forms/)
- [ ] [mwf_plots](https://beta.lmfdb.org/api/mwf_plots/)
- [ ] [mwf_tables](https://beta.lmfdb.org/api/mwf_tables/)
- [ ] [mwfp_forms](https://beta.lmfdb.org/api/mwfp_forms/)
- [ ] [nf_fields](https://beta.lmfdb.org/api/nf_fields/)
- [ ] [sl2z_subgroups](https://beta.lmfdb.org/api/sl2z_subgroups/)
- [ ] [smf_dims](https://beta.lmfdb.org/api/smf_dims/)
- [ ] [smf_ev](https://beta.lmfdb.org/api/smf_ev/)
- [ ] [smf_families](https://beta.lmfdb.org/api/smf_families/)
- [ ] [smf_fc](https://beta.lmfdb.org/api/smf_fc/)
- [ ] [smf_samples](https://beta.lmfdb.org/api/smf_samples/)

**Comment by JohnCremona** [comment]
I have finished with the ec_* tables.  Some comments:
   - ec_curves has not been used for some time and can be dropped (the data for EC/Q is now divided among several other tables)
   - ec_nfcopy appears to be a copy once made of some version of ec_nfcurves, and canbe dropped
   - ec_iqflabels cannot be dropped now since it is used by the function convert_IQF_label which is imported in several places, BUT it is many years since we had the old labels foe ideals in IQFs (imaginary quadratic fields) and I am pretty sure that whenver this function is called on a string it returns the same string, so it could be excised throughout.  I will try that and make sure that all tests pass.
   - ec_sympow is not used anywhere.  It appears to have a row for the n'th symmetric power of each isogeny class of elliptic curves of conductor <300,000 (from which one could estimate a date for when it was created), for n up to 8, and the only data it contains is the conductor of that power.  I am pretty sure that it can be dropped, as it is not used and I doubt if those conductors would be hard to compute.

**Comment by edgarcosta** [comment]
Thanks @JohnCremona. 
- `ec_iqflabels` is used to convert some old links stored in the L-functions database to the new links.
- `ec_sympow` is my own creation. Not used in the LMFDB code base, easy to generate with Magma, and used to know what L-functions for symmetric powers we will want to compute.

**Comment by JohnCremona** [comment]
OK, thanks.  Would it make sense to make a one-off change to the stored URLs so we don't need to use the function at all?  In any case, I can remove its use from ecnf/ and genus2_curves/ and leave a comment saying where it is still used.

**Comment by AndrewVSutherland** [comment]
I just dropped mf_temp and ec_nfopy (and removed them from the list above).  @edgarcosta where is ec_iqflabels used, there are no references in the current LMFDB codebase.

If anyone else knows of tables on the list above that can be dropped, please say so.  I believe the mwf* files can all be dropped (they have been replaced by the maass_* files), but two of them are still referenced in test_api.py.

**Comment by edgarcosta** [comment]
@JohnCremona, I think we can. Can you remind an example of an old label?

**Comment by JohnCremona** [comment]
> @JohnCremona, I think we can. Can you remind an example of an old label?

(I had to look this up, but there is an explanation in ecnf/WebEllipticCurve.py in a comment above the function definition.)

'65.18.1' or '[65.18.1]'  are converted to '65.2'  (from an HNF representation to the norm.index label, which depends on us having fixed a sorting order for ideals).

As it happens, until now my C++ code for producing BMFs and hence ECNFs/IQFs only used the HNF-style labels as I had not implemented the ideal sorting, so my scripts for converting the C++ output into upload files had to do this conversion anyway -- but as of today my bianchi-progs code now handles the new ideal labels itself so the old ones will soon be a distant memory.

**Comment by edgarcosta** [comment]
Tks! I think on the L-functions side we already converted every old label to a new label. So, my approach would be to remove the function, and see if we missed something.

**Comment by edgarcosta** [comment]
@AndrewVSutherland, I think `belyi_galmaps` can be dropped. @SamSchiavone, please confirm.

**Comment by JohnCremona** [comment]
> Tks! I think on the L-functions side we already converted every old label to a new label. So, my approach would be to remove the function, and see if we missed something.

I'm about to make a PR which does this mostly, though it still leaves the function there and keeps its use in utils/names_and_urls and in the Lfunctions database code.

**Comment by SamSchiavone** [comment]
@edgarcosta It should be okay. I think I switched everything over to belyi_galmaps_prim in this commit: https://github.com/LMFDB/lmfdb/pull/4634/commits/bf3ac020ab5be56465364f8112ca0d9dad31b211

**Comment by jenpaulhus** [comment]
The three hgcwa_* above are done now.  They will be replaced "soon-ish" by new tables: hgcwa_genvectors, hgcwa_per_group_stats, and hgcwa_per_genus_stats.

**Comment by AndrewVSutherland** [comment]
@roed314 I think we are ready to close this.  The only pending item we discussed is making the table description an editable knowl -- I can make a separate issue for that if you like.

**Comment by roed314** [comment]
Sure, a separate issue seems reasonable.
